### ISSUE TITLE
should `oce.as.POSIXlt()` be removed?

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1133]
Clark and I have had some discussions about `lubridate`, and so I thought that maybe I should check to see if `lubridate` functions could help with inferring dates from data files. Some grepping told me that `oce.as.POSIXlt()`, which is in `R/oce.R`, does some conversions of strings to times. So I thought maybe that should use a `lubridate` function. However, another grep tells me that `oce.as.POSIXlt()` is not called by any `oce` functions! I wonder if `oce.as.POSIXlt()` should simply be deprecated, with some advice in its manpage about how to use some other `lubridate` function instead. However, I'm damned if I can figure out which function can handle the range of dates we run into. Below are some notes. 
```R
> as_datetime("2010-10-03")
[1] "2010-10-03 03:00:00 UTC"
> as_datetime("2010-jan-03")
Error in as.POSIXlt.character(x, tz, ...) : 
  character string is not in a standard unambiguous format
> as_datetime("2010 jan 03")
Error in as.POSIXlt.character(x, tz, ...) : 
  character string is not in a standard unambiguous format
> as_datetime("2010 jan 3")
Error in as.POSIXlt.character(x, tz, ...) : 
  character string is not in a standard unambiguous format
> as_datetime("2010 Jan 3")
Error in as.POSIXlt.character(x, tz, ...) : 
  character string is not in a standard unambiguous format
> as_datetime("2010 January 3")
Error in as.POSIXlt.character(x, tz, ...) : 
  character string is not in a standard unambiguous format
```

**Comment by dankelley** [comment]
Oh, I see that there's another function that is *very* nice:
```R
library(lubridate)
x <- c("09-01-01", "09-01-02", "2009 jan 03", "3 jan 2009")
parse_date_time(x, c("ymd","dmy"))
```
yields
```
[1] "2009-01-01 UTC" "2009-01-02 UTC" "2009-01-03 UTC" "2009-01-03 UTC"
```

**Comment by richardsc** [comment]
I forgot to add to this, but I can't think of a single case that I've used `oce.as.POSIXlt()`, so I'm fine with deprecation. I agree that I don't think we should make oce depend on lubridate right now, but if someone needs the functionality it's easy to install for their own work (thanks, R packages!)

**Comment by dankelley** [comment]
Fixed in `develop` commit 4af1c975a2a278ec6a6564f3d037d329171c1c88
